arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1146 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Operational InsightsManagement Client for LinkTargets, StorageInsightConfigs and SavedSearches.",
+    "description": ".Net client wrapper for the REST API for Azure Operational Insights",
+    "version": "2015-03-20"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
+  ],
+  "produces": [
+    "application/json",
+    "text/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/storageInsightConfigs/{name}": {
+      "put": {
+        "tags": [
+          "StorageInsights"
+        ],
+        "operationId": "StorageInsights_CreateOrUpdate",
+        "description": "Create or update a storage insight.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
```

Comment:
Actually, you can now make a global parameter that is not tied to the client, but used per-method.

If you make a global parameter, and use `"x-ms-parameter-location": "method"` this will cut down on the duplication. (and not tie it to the service client class itself!)
``` json
 "ResourceGroupNameParameter": {
      "name": "resourceGroupName",
      "in": "path",
      "required": true,
      "type": "string",
      "description": "The name of the resource group within the current subscription. You can obtain this value from the Azure Resource Manager API or the portal.",
      "x-ms-parameter-location": "method"
    },
```

---
In reply to: [86867193](https://github.com/Azure/azure-rest-api-specs/pull/662#discussion_r86867193) [](ancestors = 86867193)